docs/rewards/managing-rewards.md

```diff
@@ -0,0 +1,115 @@
+
+`contract_address` - The CosmWasm address of the contract 
+
+`owner_address` - The address of the contract's owner. This field can be both an account or contract address.
+
+`rewards_address` - The account address that will receive the contract's rewards. Tokens are sent to this address after a [withdrawal function](#withdrawing-awards) is executed. If not set, the contract will not receive rewards. 
+
+**All above addresses are bech32-encoded.**
+
+### Setting The Contract Metadata 
+
+The contract metadata must be set up by using the `MsgSetContractMetadata` transaction which must be signed by a contract admin. 
+
+```go  
+message ContractMetadata { 
+  string contract_address =  "archway14hj2tavq8fpesdwxxcu44rty3hh90vhujrvcmstl4zr3txmfvw9sy85n2u";
+  string owner_address = "archway12reqvcenxgv5s7z96pkytzajtl4lf2epyfman2";
+  string rewards_address = "archway12reqvcenxgv5s7z96pkytzajtl4lf2epyfman2";
+}
+``` 
+### Updating the Contract Metadata 
+The contract metadata of a contract can be updated by the `owner_address`. This could either be a contract admin or the address of the contract itself. Updates are completed by using the `MsgSetContractMetadata` transaction. 
+
+If the `owner_address` and `contract_address` are the same, updates to the metadata can be done by using the WASM bindings. 
+
+
+
+## Reward Storage
+The rewards owed to a certain contract are stored on the blockchain. To receive these rewards, a [withdrawal operation](#withdrawing-awards) must be executed. Developers can also track these rewards by using the [RewardsRecord object](#tracking-reward-amounts). 
+
+## Tracking Reward Amounts 
+The RewardsRecord object is used to track the rewards that have been calculated for a specific `rewards_address`. 
+
+### Example Request - RewardsRecord 
+```bash
+archwayd q rewards rewards-records "${rewards_address}" --node "${node_url}" --output json | jq .
+```
+### Example Response 
+
+```json
+{
+  "id": "1",
+  "rewards_address": "archway12reqvcenxgv5s7z96pkytzajtl4lf2epyfman2",
+  "rewards": [
+    {
+      "denom": "uarch",
+      "amount": "10000"
+    }
+  ],
+  "calculated_height": 100,
+  "calculated_time": {
+    "seconds": 1660591975,
+    "nanos": 0
+  }
+}
+```
+`id` - Each Reward Record has a unique ID. This can be used to only withdraw certain reward records. 
+
+`rewards_address` - The account address that will receive the contract's rewards. Tokens are sent to this address after a [withdrawal function](#withdrawing-awards) is executed. If not set, the contract will not receive rewards. 
+
+`denom`- The token type/denmoination. uarch is the default token of the Archway Network.  
+
+`amount`- The calculated reward amount. 
+
+`calculated_height` - The block height of the rewards when the rewards were calculated. 
+
+`calculated_time` - The block time of when the rewards were calculated. 
```

Comment:
```suggestion
`id` - Each Reward Record has a unique ID. This can be used to withdraw only certain reward records.

`rewards_address` - The account address that will receive the contract's rewards. Tokens are sent to this address after executing a [withdrawal function](#withdrawing-awards). If not set, the contract will not receive rewards. 

`denom`- The token type/denomination. `uarch` is the default token of the Archway Network. The testnets use other denominators, like `uconst` and `utitus`.

`amount`- The calculated reward amount. 

`calculated_height` - The block height of the rewards when the rewards were calculated. 

`calculated_time` - The block time of when the rewards were calculated. 
```